geckordp/firefox.py

```diff
@@ -125,14 +131,19 @@ def start(url: str,
         if (override_firefox_path == ""):
             override_firefox_path = Firefox.get_binary_path()
 
-        args = [override_firefox_path,
-                "-new-instance",
-                "-no-remote",
-                "-no-default-browser-check",
-                "-new-window", url,
-                "-p", profile,
-                "--start-debugger-server", str(port)
-                ]
+        args = [
+            override_firefox_path,
+            "-new-instance",
+            "-no-remote",
+            "-no-default-browser-check",
+            "-p", profile,
+            "--start-debugger-server", str(port),
+            "-new-window",
+        ]
+        # NOTE: don't open a first page unless explicitly passed by
+        # caller.
+        if url:

```

Comment:
This part should be moved after "if (append_args is not None):" block because the url is a position dependent argument.